packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2291,29 +2302,52 @@ const completeSegmentScript1Full = stringToPrecomputedChunk(
 );
 const completeSegmentScript1Partial = stringToPrecomputedChunk('$RS("');
 const completeSegmentScript2 = stringToPrecomputedChunk('","');
-const completeSegmentScript3 = stringToPrecomputedChunk('")</script>');
+const completeSegmentScriptEnd = stringToPrecomputedChunk('")</script>');
+
+const completeSegmentData1 = stringToPrecomputedChunk(
+  '<div hidden data-react-instr="$RS" data-rarg0="',
+);
+const completeSegmentData2 = stringToPrecomputedChunk('" data-rarg1="');
+const completeSegmentDataEnd = dataElementQuotedEnd;
 
 export function writeCompletedSegmentInstruction(
   destination: Destination,
   responseState: ResponseState,
   contentSegmentID: number,
 ): boolean {
-  writeChunk(destination, responseState.startInlineScript);
-  if (!responseState.sentCompleteSegmentFunction) {
-    // The first time we write this, we'll need to include the full implementation.
-    responseState.sentCompleteSegmentFunction = true;
-    writeChunk(destination, completeSegmentScript1Full);
+  const scriptFormat =
+    !enableFizzExternalRuntime || responseState.streamingFormat === 'SCRIPT';
+  if (scriptFormat) {
+    writeChunk(destination, responseState.startInlineScript);
+    if (!responseState.sentCompleteSegmentFunction) {
+      // The first time we write this, we'll need to include the full implementation.
+      responseState.sentCompleteSegmentFunction = true;
+      writeChunk(destination, completeSegmentScript1Full);
+    } else {
+      // Future calls can just reuse the same function.
+      writeChunk(destination, completeSegmentScript1Partial);
+    }
   } else {
-    // Future calls can just reuse the same function.
-    writeChunk(destination, completeSegmentScript1Partial);
+    writeChunk(destination, completeSegmentData1);
```

Comment:
I was imagining a slightly different approach. This essentially swaps out the script tag for a hidden div tag which is fine I suppose but slightly more bytes and it probably isn't cleaning up the extra hidden div (I haven't checked yet so maybe this is wrong).

Instead I was thinking we would have a variant of `flushSegmentContainer` which if we were using the `'DATA'` runtime would decorate the segment div itself with these instruction data attributes. It's one less div and the cleanup is automatic b/c these hidden divs are removed from the document when their contents are moved to the placeholder

Reply:
Thinking more about this, it'll be harder to do for completed boundaries and it makes the code diverge more so it might be harder to see errors in the future. I think keeping this as is makes sense but maybe the runtime script that encounters these instruction divs can remove them from the dom

Reply:
> I was imagining a slightly different approach. This essentially swaps out the script tag for a hidden div tag which is fine I suppose but slightly more bytes and it probably isn't cleaning up the extra hidden div (I haven't checked yet so maybe this is wrong).
> 
> Instead I was thinking we would have a variant of `flushSegmentContainer` which if we were using the `'DATA'` runtime would decorate the segment div itself with these instruction data attributes. It's one less div and the cleanup is automatic b/c these hidden divs are removed from the document when their contents are moved to the placeholder

Thanks for the review!
Is this approach similar to what Seb mentioned in a [previous comment](https://github.com/facebook/react/pull/25437#discussion_r990539187)? Happy to keep as is (and change the MutationObserver to remove these divs from the dom), just want to understand